dask_ml/model_selection/_search.py

```diff
@@ -286,16 +304,50 @@ def normalize_params(params):
 
     return fields, tokens, params2
 
-
-def _get_fit_params(cv, fit_params, n_splits):
-    if not fit_params:
-        return [(n, None) for n in range(n_splits)]
+def _generate_fit_params_key_vals(fit_params, keys_filtered=None):
     keys = []
     vals = []
     for name, (full_name, val) in fit_params.items():
+        if keys_filtered is not None and name not in keys_filtered:
+            continue
         vals.append(val)
         keys.append((name, full_name))
-    return [(n, (cv_extract_params, cv, keys, vals, n)) for n in range(n_splits)]
+    return keys, vals
+
+def _get_weights_source(fit_params):
+    '''
+        _get_weights_source returns the weights source. sklearn pipelines subscript the parameter names such that a "samples_weight" parameter must be resolved
+        to "clf__samples_weight" for some classifier "clf" in an sklearn pipeline. Also support eval_sample_weight as a priority source.
+    '''
+    eval_weight_source = None
+    if fit_params is None:
+        return eval_weight_source
+
+    # sklearn subscripting adds __ scores to step variables
+    for candidate_param in fit_params:
+        if "eval_sample_weight" in candidate_param:
+            eval_weight_source = candidate_param
+            break
+        elif "sample_weight" in candidate_param:
+            eval_weight_source = candidate_param
+            break
+    return eval_weight_source
```

Comment:
```
candidate_set = set(candidate_param.keys())
return  "eval_sample_weight" in candidate_set or "sample_weight" in candidate_set or None
```
you may simplify this function like above

Reply:
will add in next commit.

Reply:
```
candidate_set = set(candidate_param.keys())
if "eval_sample_weight" in candidate_set:
    return "eval_sample_weight"
elif "sample_weight" in candidate_set:
    return "sample_weight"
else:
    return None
```
the original code will return bool or None, this one returns str or None

Reply:
after looking at this it won't work since it won't give the right value for a pipeline

In a pipeline like `Pipeline([("clf", LogisticRegression())])` the `sample_weight` for the logistic regression will be `clf__sample_weight`  so it has to return  `clf__sample_weight` when it is that param